bundle/config/loader/process_include_test.go

```diff
@@ -33,3 +38,339 @@ func TestProcessInclude(t *testing.T) {
+			Locations: []dyn.Location{
+				{File: filepath.FromSlash("testdata/format/foo.job.yml"), Line: 4, Column: 7},
+				{File: filepath.FromSlash("testdata/format/foo.job.yml"), Line: 7, Column: 7},
+			},
+			Paths: []dyn.Path{
+				dyn.MustPathFromString("resources.jobs.bar"),
+				dyn.MustPathFromString("resources.jobs.foo"),
+			},
+		},
+	}, diags)
+}
+
+func TestResourceNames(t *testing.T) {
+	names := []string{}
+	typ := reflect.TypeOf(config.Resources{})
+	for i := 0; i < typ.NumField(); i++ {
+		field := typ.Field(i)
+		jsonTags := strings.Split(field.Tag.Get("json"), ",")
+		singularName := strings.TrimSuffix(jsonTags[0], "s")
+		names = append(names, singularName)
+	}
+
+	// Assert the contents of the two lists are equal. Please add the singular
+	// name of your resource to resourceNames global if you are adding a new
+	// resource.
+	assert.Equal(t, len(resourceTypes), len(names))
+	for _, name := range names {
+		assert.Contains(t, resourceTypes, name)
+	}
+}
+
+func TestValidateFileFormat(t *testing.T) {
+	onlyJob := config.Root{
+		Resources: config.Resources{
+			Jobs: map[string]*resources.Job{
+				"job1": {},
+			},
+		},
+		Targets: map[string]*config.Target{
+			"target1": {
+				Resources: &config.Resources{
+					Jobs: map[string]*resources.Job{
+						"job1": {},
+					},
+				},
+			},
+		},
+	}
+	onlyJobBundle := bundle.Bundle{Config: onlyJob}
+
+	onlyPipeline := config.Root{
+		Resources: config.Resources{
+			Pipelines: map[string]*resources.Pipeline{
+				"pipeline1": {},
+			},
+		},
+	}
+	onlyPipelineBundle := bundle.Bundle{Config: onlyPipeline}
+
+	bothJobAndPipeline := config.Root{
+		Resources: config.Resources{
+			Jobs: map[string]*resources.Job{
+				"job1": {},
+			},
+		},
+		Targets: map[string]*config.Target{
+			"target1": {
+				Resources: &config.Resources{
+					Pipelines: map[string]*resources.Pipeline{
+						"pipeline1": {},
+					},
+				},
+			},
+		},
+	}
+	bothJobAndPipelineBundle := bundle.Bundle{Config: bothJobAndPipeline}
+
+	twoJobs := config.Root{
+		Resources: config.Resources{
+			Jobs: map[string]*resources.Job{
+				"job1": {},
+				"job2": {},
+			},
+		},
+	}
+	twoJobsBundle := bundle.Bundle{Config: twoJobs}
+
+	twoJobsTopLevelAndTarget := config.Root{
+		Resources: config.Resources{
+			Jobs: map[string]*resources.Job{
+				"job1": {},
+			},
+		},
+		Targets: map[string]*config.Target{
+			"target1": {
+				Resources: &config.Resources{
+					Jobs: map[string]*resources.Job{
+						"job2": {},
+					},
+				},
+			},
+		},
+	}
+	twoJobsTopLevelAndTargetBundle := bundle.Bundle{Config: twoJobsTopLevelAndTarget}
+
+	twoJobsInTarget := config.Root{
+		Targets: map[string]*config.Target{
+			"target1": {
+				Resources: &config.Resources{
+					Jobs: map[string]*resources.Job{
+						"job1": {},
+						"job2": {},
+					},
+				},
+			},
+		},
+	}
+	twoJobsInTargetBundle := bundle.Bundle{Config: twoJobsInTarget}
```

Comment:
These can all be files under `testdata`. These tests now use a mix of files and inline.

Reply:
converted all the tests to files.